### ISSUE TITLE
If there are no snapshots in a repository, return `[]` from restic snapshots --json

### ISSUE THREAD

**Issue by chrahunt** [restic/restic#2979]
Output of `restic version`
--------------------------

restic 0.9.6 (v0.9.6-204-gf0b2dee7) compiled with go1.12.9 on linux/amd64

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------

I ran this

```
cd "$(mktemp -d)"
mkdir repo
RESTIC_REPOSITORY="local://$PWD/repo"
RESTIC_PASSWORD=a
restic init
restic --json snapshots
```

`restic --json snapshots` printed `null`, but I would have expected `[]`. I think using `[]` would make the output easier to process correctly.

What are you trying to do? What problem would this solve?
---------------------------------------------------------

I use restic to back up an explicit list of files. When this list of files changes, restic is not able to locate the parent snapshot. This is documented in #2246.

To work around this I am using a variation of [#2246 (comment)](https://github.com/restic/restic/issues/2246#issuecomment-628022549):

```
restic backup --parent "$(restic  --json snapshots | jq -r 'max_by(.time) | .short_id // empty')"
```

This works, but when the repository does not have a snapshot `jq` traces an error message: `jq: error (at <stdin>:1): Cannot iterate over null (null)`. Printing `[]` when there are no snapshots would make this obvious approach the best approach.

Sorting output is another case when the straightforward approach is wrong by default. Normally I would do

```
restic snapshots --json | jq 'sort_by(.time) | reverse'
```

but this fails for new repositories for the same reason as above.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

restic considered all my files new whenever any one was added, but the impact was just extra CPU consumed on my local PC instead of reuploading files to the cloud storage provider. That's a very safe way to fallback, and I'm happy I switched. :)

**Comment by rawtaz** [comment]
So, restic uses [these rules](https://golang.org/pkg/encoding/json/#Marshal) to encode the list of snapshots to JSON, and this by default turns an empty list which you expect to be `[]` into `null` like you observed.

A simple workaround for this is to pipe in `sed 's/^null$/[]/'` before your `jq` command, so it becomes `restic backup --parent "$(restic  --json snapshots | sed 's/^null$/[]/ | jq -r 'max_by(.time) | .short_id // empty')" `. This will only replace `null` if it's the only thing on a line without whitespace around it, and I find it unlikely that this will match anything else than what you want to target.

Since we haven't seen any other requests for this (AFAIK) and the workaround is simple, I suggest that for now we don't change the JSON output (which would technically be a BC change), and instead use this workaround. Thoughts?

**Comment by rawtaz** [comment]
After discussing with @fd0 we consider this a bug and it's fixed in #2984.

**Comment by chrahunt** [comment]
Awesome, thank you!
